podcasts/Onboarding/Plus/UpgradeLandingView.swift

```diff
@@ -0,0 +1,324 @@
+
+                            FeaturesCarousel(currentIndex: $currentPage.animation(), currentPrice: $displayPrice, tiers: tiers)
+
+                            PageIndicator(numberOfItems: tiers.count, currentPage: currentPage)
+                            .padding(.top, 27)
+                        }
+                    }
+                }
+            }
+        }
+        .background(Color(hex: "121212"))
+    }
+
+    var topBar: some View {
+        HStack(spacing: 0) {
+            Rectangle()
+                .foregroundColor(.clear)
+                .frame(height: 44)
+            Button(viewModel.source == .upsell ? L10n.eoyNotNow : L10n.plusSkip) {
+                viewModel.dismissTapped()
+            }
+            .foregroundColor(.white)
+            .font(style: .body, weight: .medium)
+            .padding(.trailing)
+        }
+    }
+
+    enum DisplayPrice {
+        case yearly, monthly
+    }
+}
+
+// MARK: - Feature Carousel
+
+struct FeaturesCarousel: View {
+    let currentIndex: Binding<Int>
+
+    let currentPrice: Binding<UpgradeLandingView.DisplayPrice>
+
+    let tiers: [UpgradeTier]
+
+    @State var calculatedCardHeight: CGFloat?
+
+    var body: some View {
+        // Store the calculated card heights
+        var cardHeights: [CGFloat] = []
+
+        HorizontalCarousel(currentIndex: currentIndex, items: tiers) {
+            UpgradeCard(tier: $0, currentPrice: currentPrice)
+                .overlay(
+                    // Calculate the height of the card after it's been laid out
+                    GeometryReader { proxy in
+                        Action {
+                            // Add the calculated height to the array
+                            cardHeights.append(proxy.size.height)
+
+                            // Determine the max height only once we've calculated all the heights
+                            if cardHeights.count == tiers.count {
+                                calculatedCardHeight = cardHeights.max()
+
+                                // Reset the card heights so any view changes won't use old data
+                                cardHeights = []
+                            }
+                        }
+                    }
+                )
+        }
+        .carouselPeekAmount(.constant(Constants.peekAmount))
+        .carouselItemSpacing(Constants.spacing)
+        .frame(height: calculatedCardHeight)
+    }
+
+    private enum Constants {
+        static let peekAmount: Double = 20
+        static let spacing: Double = UIDevice.current.isiPad() ? 150 : 30
+    }
+}
+
+// MARK: - Available plans
+
+struct UpgradeTier: Identifiable {
+    let tier: Tier
+    let iconName: String
+    let title: String
+    let yearlyIdentifier: String
+    let monthlyIdentifier: String
+    let description: String
+    let buttonLabel: String
+    let buttonColor: Color
+    let buttonForegroundColor: Color
+    let features: [TierFeature]
+    let background: AnyView
```

Comment:
We can avoid the AnyView by just specifying this as a LinearGradient instead. 

```suggestion
    let background: LinearGradient
```

Reply:
Done in 7ad7ff6485963cfde5da16bbcfbf35fbaa2346dc